File: app/src/main/java/com/hover/stax/onboarding/SlidesFragment.java
package com.hover.stax.onboarding;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;


public class SlidesFragment extends Fragment {
	private static final String TAG = "SlidesFragment";
	private int resLayout = 0;

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		if (getArguments() != null)
			resLayout = getArguments().getInt(TAG, 0);
	}

	public static SlidesFragment newInstance(int resLayout) {
		SlidesFragment fragment = new SlidesFragment();
		Bundle args = new Bundle();
		args.putInt(TAG, resLayout);
		fragment.setArguments(args);
		return fragment;
	}

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
		return inflater.inflate(resLayout, container, false);
	}
}

File: app/src/main/java/com/hover/stax/onboarding/SlidesFragment.kt
package com.hover.stax.onboarding

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment

class SlidesFragment : Fragment() {

    private var resLayout: Int? = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        arguments?.let { resLayout = arguments?.getInt(TAG, 0) }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(resLayout!!, container, false)
    }

    companion object {
        val TAG: String = SlidesFragment::class.java.simpleName

        fun newInstance(resLayout: Int): SlidesFragment = SlidesFragment().apply {
            arguments = bundleOf(TAG to resLayout)
        }
    }
}
